app/javascript/article-form/articleForm.jsx

```diff
@@ -343,6 +356,30 @@ export default class ArticleForm extends Component {
           helpPosition={helpPosition}
           version={version}
         />
+        {this.state.showModal && (
+          <Modal
+            size="s"
+            title="You have unsaved changes"
+            onClose={this.toggleModal}
+          >
+            <p>
+              You've made changes to your post. Do you want to navigate to leave
+              this page?
+            </p>
+            <div className="pt-4 flex gap-2">
+              <Button variant="danger" url="/" tagName="a">
+                Yes, leave the page
+              </Button>
+              <Button
+                aria-label="Cancel"
+                variant="secondary"
+                onClick={this.toggleModal}
+              >
+                No, keep editing
+              </Button>
+            </div>
```

Comment:
(refactor / blocking) - In Safari we're missing the spacing between the buttons as the `grid-gap` isn't supported with `flex` :(
Maybe we can change the container to a `grid` layout with two columns instead?

![screenshot of modal in safari, with no gap between the buttons](https://user-images.githubusercontent.com/20773163/105976374-ff006700-6087-11eb-9de4-892397287c68.png)

Reply:
OH DAMN! #TIL it's not supported in Safari ://// I think we've been all using combination of `flex` and `gap-*` in other places too which means........ THINGS.... ARE.... BROKEN in Safari... DAMN IT.

---

In that case, I'd suggest going with `mr-2` for the first button. I live by the rule: _[use `grid` for layouts and `flex` for everything else](https://css-tricks.com/grid-for-layout-flexbox-for-components/)_.

Reply:
https://github.com/forem/forem/issues/12452